hamp_radar/iquick.py

```diff
@@ -116,15 +244,41 @@ def compact_geometry(
 
 
 def extract_raw_arrays(data, mmbgs: Iterable[MultiMainBlockGeometry]):
+    """
+    Generator function to extract arrays from 'data' based on the
+    geometry given by the iterator over MultiMainBlockGeometry instances.
+
+    Iterating over a list of this generator yields a tuple for the subblocks
+    across all the MultiMainBlockGeometry instances sequentially.
+
+    Optimisation uses NumPy library as_strided function to create a view of the
+    original data array interpreted with different shape and strides. Shape will
+    have (nrows, ncols) = (mmbg.count, block.size). Stride will be 1 unless
+    mmbg.count > 1, in which case mmbg.step is used to advance to the next
+    required subblock (skipping past other subblocks with different tags)
+
+    Args:
+        data: The input data (memory map or open file) from which to extract the
+              arrays.
+        mmbgs (Iterable[MultiMainBlockGeometry]): An iterable of
+              MultiMainBlockGeometry instances.
+
+    Yields:
+        tuple: A tuple containing:
+            - mmbg.tag: The tag/signature of the main block.
+            - block.tag: The tag/signature of the subblock.
+            - ndarray: A view of the data array corresponding to the subblock,
+                       created using numpy's as_strided function.
```

Comment:
Again, it's an implementation detail, which probably doesn't help outside users.
```suggestion
            - ndarray: A view of the data array corresponding to the subblock.
```